src/encryption.rs

```diff
@@ -1,27 +1,35 @@
-// Copyright 2019 MaidSafe.net limited.
+// Copyright 2021 MaidSafe.net limited.
 //
 // This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
 // Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
 // under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-use crate::sequential::{Iv, Key};
-use crate::SelfEncryptionError;
+use crate::Error;
 use aes::Aes128;
-use block_modes::block_padding::Pkcs7;
-use block_modes::{BlockMode, Cbc};
+use block_modes::{block_padding::Pkcs7, BlockMode, Cbc};
+use bytes::Bytes;
+use xor_name::XOR_NAME_LEN;
+
 type Aes128Cbc = Cbc<Aes128, Pkcs7>;
 
-pub const KEY_SIZE: usize = 16;
-pub const IV_SIZE: usize = 16;
+pub(crate) const KEY_SIZE: usize = 16;
+pub(crate) const IV_SIZE: usize = 16;
+
+pub(crate) const HASH_SIZE: usize = XOR_NAME_LEN;
+pub(crate) const PAD_SIZE: usize = (HASH_SIZE * 3) - KEY_SIZE - IV_SIZE;
+
+pub(crate) struct Pad(pub [u8; PAD_SIZE]);
+pub(crate) struct Key(pub [u8; KEY_SIZE]);
+pub(crate) struct Iv(pub [u8; IV_SIZE]);
```

Comment:
What is `Iv` here? Perhaps we can rename/expand the acronym for clarity?

It's not immediately clear what `Pad` is all about here either tbh. Perhaps can you add a wee comment for each?

Reply:
Ah, this is crypto lingo. Let's see if there's some commonly used variations, but at least some docs would be nice.

Reply:
IV: https://en.wikipedia.org/wiki/Initialization_vector
Padding: https://en.wikipedia.org/wiki/Padding_(cryptography)

Reply:
I think these are pretty standard names.

Reply:
It's funny, I have added the docs, but this comment doesn't show as outdated, and in the changes view, the docs doesn't show..